src/runner.cpp

```diff
@@ -31,11 +35,57 @@
 namespace libsemigroups {
   namespace py = pybind11;
 
+  namespace {
+    using std::chrono::high_resolution_clock;
+    using std::chrono::system_clock;
+    using std::chrono::time_point_cast;
+    template <
+        typename TimePoint,
+        std::enable_if_t<std::is_same_v<TimePoint, high_resolution_clock>, bool>
+        = true>
+    system_clock::time_point to_system(TimePoint tp) {
+      return tp;
+    }
+
+    template <
+        typename TimePoint,
+        std::enable_if_t<!std::is_same_v<TimePoint, high_resolution_clock>,
+                         bool>
+        = true>
+    system_clock::time_point to_system(TimePoint tp) {
+      // Account for the difference between system_clock and
+      // high_resolution_clock
+      auto sys_now      = system_clock::now();
```

Comment:
I'm not sure this does what you want, the time at which line 58 is executed isn't the same as the time at which 59 is executed, so what are you measuring the here?